app/components/file_attachment_list/file_attachment_list.js

```diff
@@ -69,13 +69,11 @@ export default class FileAttachmentList extends PureComponent {
     }
 
     componentDidUpdate(prevProps) {
-        if (prevProps.files !== this.props.files) {
-            this.filesForGallery = this.getFilesForGallery(this.props);
-            this.buildGalleryFiles().then((results) => {
-                this.galleryFiles = results;
-            });
-        }
-        if (this.props.files !== prevProps.files && this.props.files.length === 0) {
+        this.filesForGallery = this.getFilesForGallery(this.props);
```

Comment:
Given that `prevProps.files !== this.props.files` always returns true because the comparison between two arrays, even if they are empty:
```js
> [1] === [2]
false
> [1] === [1]
false
> [] === []
false
```
this part of the code is being excuted every time we update the component so I've extracted it from the conditional